### PR TITLE
Change 'bool required' to enum

### PR DESCRIPTION
**Pull request by agentydragon**

### CODE REVIEW

File: args.h [hunk 1]
```diff
@@ -175,28 +175,26 @@ typedef internal::EnumFlag Enum;
...
 typedef internal::BoolFlag Bool;
 
-// Constants used to make code using args framework more clear.
-// This may be possibly changed to an enum if there is need for more
-// "flag types" than REQUIRED and OPTIONAL.
-const bool REQUIRED = true;
-const bool OPTIONAL = false;
+// Denotes whether a flag is required or optional.
+// TODO: Very awkward name.
+enum Requiredness { REQUIRED, OPTIONAL };
```
Comment (bageta) [inline args.h]: I don't like name Requiredness. What about Mandatory or something like that?

Comment (agentydragon) [inline args.h]: Me neither, but "Mandatory" is no good either. "Mandatory" is the same thing as "required" - I would expect `enum X { MANDATORY, OPTIONAL };`, not `enum Mandatory { X, OPTIONAL };` (it's a bit like `enum Red { RED, GREEN };`).

Maybe "Mandatoriness"? "Optionality"?

Comment (bageta) [inline args.h]: In XML they use something like this: "use=optional|prohibited|required" for attributes. I think that sounds good.

But I'm ok with "Optionality" too.

Comment (agentydragon) [inline args.h]: How about 'FlagUse'? (I'm not sure if 'Use' alone expresses the intent clearly enough.)

**Comment by bageta** [comment]
Sounds good.

File: args.h (hunk 1 above)
Comment (bageta) [inline args.h]: Sounds good.

**Comment by agentydragon** [comment]
Done.

**Comment by bageta** [comment]
ok

**Pull request merged**